docs/x509/tutorial.rst

```diff
@@ -82,3 +82,70 @@ a few details:
     ...     f.write(csr.public_bytes(serialization.Encoding.PEM))
 
 Now we can give our CSR to a CA, who will give a certificate to us in return.
+
+Creating a self-signed certificate
+----------------------------------
+
+While most of the time you want a certificate that has been *signed* by someone
+else (i.e. a certificate authority), so that trust is established, sometimes
+you want to create a self-signed certificate. Self-signed certificates are not
+issued by a certificate authority, but are instead signed by themselves.
```

Comment:
The certificate doesn’t sign itself.  You create a key first that you use for signing, no?